src/Tizen.NUI/src/public/Capture.cs

```diff
@@ -0,0 +1,252 @@
+        /// <param name="size"></param>
+        /// <param name="path"></param>
+        /// <param name="color"></param>
+        public void Start(Container source, Size size, string path, Color color)
+        {
+            if (source is View || source is Layer)
+            {
+                Interop.Capture.Start1(swigCPtr, source.SwigCPtr, new Vector2(size.Width, size.Depth).SwigCPtr, path, new Vector4(color.R, color.G, color.B, color.A).SwigCPtr);
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            }
+        }
+
+        /// <summary>
+        /// Start
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="size"></param>
+        /// <param name="path"></param>
+        public void Start(Container source, Size size, string path)
+        {
+            Tizen.Log.Debug("NUI", $"Start() path={path}");
+
+            if (source is View || source is Layer)
+            {
+                Tizen.Log.Debug("NUI", $"source swigcptr={source.SwigCPtr:X}, source handle={source.SwigCPtr.Handle}");
+                Interop.Capture.Start2(swigCPtr, source.SwigCPtr, new Vector2(size.Width, size.Depth).SwigCPtr, path);
+                if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            }
+        }
+
+        private void onFinished(IntPtr data, int state)
+        {
+            if (data != IntPtr.Zero)
+            {
+                var arg = new CaptureFinishStateEventArgs();
+                arg.State = (CaptureFinishedState)state;
+                finishedEventHandler?.Invoke(this, arg);
+            }
+        }
+
+        private CaptureSignal finishedSignal;
+        private delegate void finishedCallbackType(IntPtr data, int state);
+        private finishedCallbackType finishedCallback;
+        private EventHandler<CaptureFinishStateEventArgs> finishedEventHandler;
+
+        /// <summary>
+        /// Finished
+        /// </summary>
+        public event EventHandler<CaptureFinishStateEventArgs> Finished
+        {
+            add
+            {
+                if (finishedEventHandler == null && disposed == false)
+                {
+                    finishedSignal = new CaptureSignal(Interop.Capture.Get(swigCPtr), false);
+                    finishedCallback = onFinished;
+                    finishedSignal.Connect(finishedCallback);
+                }
+                finishedEventHandler += value;
+            }
+            remove
+            {
+                finishedEventHandler -= value;
+
+                if (finishedEventHandler == null && finishedSignal?.Empty() == false)
+                {
+                    finishedCallback = onFinished;
+                    finishedSignal.Disconnect(finishedCallback);
+                }
+            }
+        }
+
+        /// <summary>
+        /// CaptureFinishedState
+        /// </summary>
```

Comment:
Would we make this enum private?
We may can change Dali::Capture::FinishedState to the simple bool value in NUI.